### PR TITLE
Additonal sorting options for tag cloud

### PR DESCRIPTION
**Pull request by alan-null**
Issue: #150

### CODE REVIEW

File: Website/Components/TagCloudCore.cs [hunk 1]
```diff
@@ -28,8 +30,30 @@ public class TagCloudCore : ITagCloudCore
...
         {
+                if (m_SortingNames != null)
+                {
+                    return m_SortingNames;
+                }
+                if (string.IsNullOrEmpty(sortingOptions))
+                {
+                    return new string[0];
+                }
+                var database = Context.Database;
```
Comment (adeneys) [inline Website/Components/TagCloudCore.cs]: Avoid use of Sitecore.Context as it makes testing and reuse difficult. Instead, we've already got the CurrentBlog property, use the Database off that.

File: Website/Views/WeBlog/TagCloud.cshtml [hunk 2]
```diff
@@ -1,13 +1,33 @@
...
+@using System.Web
+@using Sitecore.Modules.WeBlog.Extensions
 @using Sitecore.Modules.WeBlog.Globalization
-@model IEnumerable<Sitecore.Modules.WeBlog.Model.TagCloudRenderingModel>
+@model Sitecore.Modules.WeBlog.Model.TagCloudRenderingModel
 
 <div class="wb-tag-cloud wb-panel">
     <h3>@Translator.Render("TAGCLOUD").ToHtmlString()</h3>
+
+    <ul class="wb-tag-sorting">
```
Comment (adeneys) [inline Website/Views/WeBlog/TagCloud.cshtml]: Could we add a parameter to the rendering to allow the user to select if they want to show the sorting options to the user or just use a single selected option?

File: Website/Components/TagCloudCore.cs [hunk 3]
```diff
@@ -28,8 +30,30 @@ public class TagCloudCore : ITagCloudCore
...
         {
+                if (m_SortingNames != null)
+                {
+                    return m_SortingNames;
+                }
+                if (string.IsNullOrEmpty(sortingOptions))
+                {
+                    return new string[0];
+                }
+                var database = CurrentBlog.Database;
```
Comment (alan-null) [inline Website/Components/TagCloudCore.cs]: Updated

**Comment by alan-null** [comment]
> Could we add a parameter to the rendering to allow the user to select if they want to show the sorting options to the user or just use a single selected option?

Only selected sorting options in rendering parameters will be rendered. 
If you select only one, then the rest will be hidden. 
image

If you want to add additional parameter to prevent options from being rendered, let me know.

**Pull request merged**